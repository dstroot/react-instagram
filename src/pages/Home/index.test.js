import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';

// component to test
import { Home } from '.';
import data from 'components/InstagramBackground/__test__/data.json';

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);

  // cleanup mocked fetch
  fetch.resetMocks();
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('Test page Home:', () => {
  it('renders', async () => {
    //arrange
    fetch.mockResponseOnce(JSON.stringify(data));

    // act (Use the asynchronous version of act to apply resolved promises)
    await act(async () => {
      render(<Home />, container);
    });

    // assert
    expect(container.innerHTML).not.toBeNull();
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });

  it('contains the expected text', async () => {
    //arrange
    fetch.mockResponseOnce(JSON.stringify(data));

    // act (Use the asynchronous version of act to apply resolved promises)
    await act(async () => {
      render(<Home />, container);
    });

    // assert
    expect(container.textContent).toContain("Boom! It's kyliecosmetics");
  });
});
