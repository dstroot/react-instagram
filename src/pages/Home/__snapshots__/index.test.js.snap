// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test page Home: renders 1`] = `
"<div class=\\"sc-htpNat kWgLAi\\">
  <div class=\\"sc-bZQynM khTqKj\\"></div>
  <div class=\\"sc-bxivhb hkLWXj\\">
    <h3 class=\\"sc-ifAKCX kDdUiS\\">Boom! It's <a href=\\"https://www.instagram.com/kyliecosmetics/\\" class=\\"sc-EHOje fZOIgH\\">kyliecosmetics</a></h3>
    <p>Thanks for visiting! Follow us on <a href=\\"https://www.instagram.com/kyliecosmetics/\\" class=\\"sc-EHOje fZOIgH\\">Instagram</a> to get the lastest looks. Check out the page background! Cool right?</p>
    <form><input name=\\"instagram\\" type=\\"text\\" required=\\"\\" class=\\"sc-bdVaJa bCSXtR\\" value=\\"kyliecosmetics\\"><button type=\\"submit\\" value=\\"Submit\\" class=\\"sc-bwzfXH hnAKSA\\">Use my Instagram!</button></form>
  </div>
</div>"
`;
